verifiers/internal/gha/builder.go

```diff
@@ -129,9 +129,28 @@ func verifyTrustedBuilderID(certPath, certTag string, expectedBuilderID *string,
 // This lets us use the pre-build builder binary generated during release (release happen at main).
 // For other projects, we only allow semantic versions that map to a release.
 func verifyTrustedBuilderRef(id *WorkflowIdentity, ref string) error {
-	if (id.CallerRepository == trustedBuilderRepository ||
-		id.CallerRepository == e2eTestRepository) &&
-		strings.EqualFold("refs/heads/main", ref) {
+	if id.CallerRepository == trustedBuilderRepository ||
+		id.CallerRepository == e2eTestRepository {
```

Comment:
This filters for the caller repositories being the e2e repo or the presubmits -- so only we can use this.

That being said, I think you have a point with the artifacts on slsa-github-generator: e.g. the builder binaries that may get built during presubmit would verify with slsa-verifier even though it uses `main`

Reply:
although, for pre-submits, the ref would be `pulls/xxx`, not main?

Reply:
when we pull the binary during a build (e.g., go builder), we (the maintainers) could swap the binaries with ones built for a tag but using @main, and it would pass verification https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/generate-builder/builder-fetch.sh#L76-L81 ... 

not sure how practical the attack is

Reply:
> although, for pre-submits, the ref would be pulls/xxx, not main?

No this is for the builder tag - so it would be at main

Reply:
> when we pull the binary during a build (e.g., go builder), we (the maintainers) could swap the binaries with ones built for a tag but using @main, 

Yes! I think this is good justification for guarding the main check.

Reply:
I added a new environment variable `SLSA_VERIFIER_TESTING` and gated these checks on it.

@asraa However, I noticed that the docker-based builder had some regression check test data that verified at `main`. I decided it wasn't a good idea to set `SLSA_VERIFIER_TESTING` for regression tests. That way we know we are testing on data for past releases properly. Probably the best thing to do for docker-based is to add unit tests for verifying at main, and regression tests for actual releases. I think we should be able to generate test data for 2.1.0 and the next release I'm about to do? wdut?

Reply:
Nevermind, I found that we have more tests that were generated at `main` so I decided I'll just set `SLSA_VERIFIER_TESTING` on regression tests for now, and track removing it on #554.